cargo-up/cargo-up-derive/src/impls/upgrader.rs

```diff
@@ -23,28 +18,13 @@ pub fn upgrader(input: &DeriveInput) -> TokenStream {
         _ => abort_call_site!("`#[upgrader]` only supports non-tuple structs"),
     };
 
-    fields.push(Field {
-        attrs: vec![],
-        vis: Visibility::Inherited,
-        ident: Some(parse_str("_upgrader").unwrap()),
-        colon_token: Some(Colon::default()),
-        ty: parse_str("::cargo_up::UpgraderInner").unwrap(),
-    });
-
-    let mut new = input.clone();
-
-    if let Data::Struct(d) = new.data {
-        new.data = Data::Struct(DataStruct {
-            fields: Fields::Named(FieldsNamed {
-                brace_token: Brace::default(),
-                named: fields,
-            }),
-            ..d
-        });
-    }
+    let upgrader = quote!(_upgrader: ::cargo_up::UpgraderInner);
```

Comment:
Lol, I forgot about this and went with `parse_str`

Reply:
You actually *had* to use `parse_str` since you needed exact types (like `syn::Type`) instead of `TokenStream`. The thing is, most of the time you don't need exact types, as this PR shows.